cranelift-codegen/meta/src/shared/immediates.rs

```diff
@@ -29,6 +29,17 @@ pub fn define() -> Vec<OperandKind> {
         .build();
     kinds.push(uimm32);
 
+    // An unsigned 128-bit immediate integer operand.
+    //
+    // This operand is used to pass entire 128-bit vectors as immediates to
+    // instructions like const.
+    let uimm128 = Builder::new_imm("uimm128")
+        .doc("A 128-bit immediate unsigned integer.")
+        .rust_type("std::boxed::Box<ir::immediates::Uimm128>")
+        .match_as_ref(true)
```

Comment:
We won't be able to fit all of the constant data (128-bit or otherwise) into `InstructionData`; I chose to box up the type and then was forced to add extra code to understand when Rust needs to use `ref` when matching.